src/SIM/SIMinput.C

```diff
@@ -678,7 +678,10 @@ bool SIMinput::parseOutputTag (const TiXmlElement* elem)
     utl::getAttribute(elem,"step",steps);
     utl::getAttribute(elem,"eps",dmp.eps);
     dmp.step.insert(steps.begin(),steps.end());
-    dmp.format = format[0];
+    if (format == "matlab")
+      dmp.format = LinAlg::MATLAB;
+    else if (format[0] == 'M' || format[0] == 'm')
+      dmp.format = LinAlg::MATLAB;
```

Comment:
This could be simplified to
```
  if (toupper(format[0]) == 'M')
    dmp.format = LinAlg::MATLAB;
```
will cover all cases above

Reply:
i do not want to cover all cases cause it would also match matrix_market. i don't like one-letter magic codes, but i need to keep it for backwards compatibility. i can change to use toupper though (old code)

Reply:
I agree, it is bad magic, and since the new format added in the subsequent commit also starts with 'm'. Then forget the backward- compatibility. The very few others that dumps the system matrices can take that change I believe (may consider at "unknown matrix format  (ignored)" warning instead"